scripts/sync_pop_candidate.sh

```diff
@@ -0,0 +1,76 @@
+
+#!/bin/bash
+
+
+cph=$(git rev-list -n 1 CHERRY_PICK_HEAD 2> /dev/null)
+
+set -o errexit
+set -o pipefail
+
+pop_all=true
+
+set +u
+while getopts 'a:' flag; do
+  case "${flag}" in
+    a) pop_all=true; shift ;;
+    \?) exit 1 ;;
+    *) echo "unexpected option ${flag}"; exit 1 ;;
+  esac
+
+  shift
+done
+set -u
+
+remote="${1:-api}"
+subtree_dir="staging/${remote}"
+cherrypick_set="${remote}.cherrypick"
+remaining=$(wc -l < "${cherrypick_set}")
+
+function pop() {
+    rc=$(head -n 1 "${cherrypick_set}")
+    if [[ ! $rc ]]; then
+        printf 'nothing to pick'
+        exit
+    fi
+    printf 'popping: %s\n' "${rc}"
+
+    if [[ ! $cph ]]; then
+        git cherry-pick --allow-empty --keep-redundant-commits -Xsubtree="${subtree_dir}" "${rc}"
+    else
+        if [[ $cph != "${rc}" ]]; then
+            printf 'unexpected CHERRY_PICK_HEAD:\ngot %s\nexpected: %s\n' "${cph}" "${rc}"
+            exit
+        fi
+        printf 'cherry-pick in progress for %s\n' "${cph}"
+        git add .
+
+        if [[ -z $(git status --porcelain) ]]; then
+            git commit --allow-empty
+        else
+            git cherry-pick --continue
+        fi
+    fi
+
+
+    # 1. Pop next commit off cherrypick set
+    # 2. Cherry-pick
+    # 3. Ammend commit
+    # 4. Remove from cherrypick set
+    make vendor
+    make manifests
+    git add .
+    git status
+    git commit --amend --allow-empty --no-edit --trailer "Upstream-repository: ${remote}" --trailer "Upstream-commit: ${rc}"
```

Comment:
Note: the --trailer command requires new git versions. I saw locally that I needed to build git from source as f34 (f35?) didn't have this in the dnf stack.

Reply:
Do you happen to know from which version --trailer was introduced? I'm having trouble finding it. I could put in a check and a link.

Reply:
IIRC it's git 2.33